backend/bin/deal-observer-backend.js

```diff
@@ -1,54 +1,63 @@
 import { createPgPool } from '@filecoin-station/deal-observer-db'
 import * as Sentry from '@sentry/node'
-import { ethers } from 'ethers'
-import assert from 'node:assert/strict'
 import timers from 'node:timers/promises'
 import slug from 'slug'
-import { RPC_URL, rpcHeaders } from '../lib/config.js'
 import '../lib/instrument.js'
-import {
-  observeBuiltinActorEvents
-} from '../lib/deal-observer.js'
 import { createInflux } from '../lib/telemetry.js'
+import { getChainHead, rpcRequest } from '../lib/rpc-service/service.js'
+import { fetchDealWithHighestActivatedEpoch, observeBuiltinActorEvents } from '../lib/deal-observer.js'
+import assert from 'node:assert'
 
 const { INFLUXDB_TOKEN } = process.env
-assert(INFLUXDB_TOKEN, 'INFLUXDB_TOKEN required')
-
+if (!INFLUXDB_TOKEN) {
+  console.error('INFLUXDB_TOKEN was not provided. Please set the INFLUXDB_TOKEN environment variable.')
+}
+const LOOP_BACK_INTERVAL = 10 * 1000
+// Filecoin will need some epochs to reach finality.
+// We do not want to fetch deals that are newer than the current chain head - 940 epochs.
+const finalityEpochs = 940
+const maxPastEpochs = 1999
+assert(finalityEpochs <= maxPastEpochs)
 const pgPool = await createPgPool()
 
-const fetchRequest = new ethers.FetchRequest(RPC_URL)
-fetchRequest.setHeader('Authorization', rpcHeaders.Authorization || '')
-const provider = new ethers.JsonRpcProvider(fetchRequest, null, { polling: true })
-
+const LOOP_NAME = 'Built-in actor events'
 const { recordTelemetry } = createInflux(INFLUXDB_TOKEN)
 
-const loop = async (name, fn, interval) => {
+const dealObserverLoop = async (makeRpcRequest, pgPool) => {
   while (true) {
     const start = Date.now()
     try {
-      await fn()
+      // If the store is empty we set the lastEpochStore to 0 and start fetching from the current chain head
```

Comment:
I'm neither seeing a variable `lastEpochStore` (do you mean `lastEpochStored`?) nor where it would be set to 0 🤔